### PR TITLE
hb-04 is done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

**Review by Borisko130 — APPROVED** [review]

File: src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    MovieSessionDao movieSessionDao;
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java]: what about access modifiers, do we want to have default ones?

File: src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.CinemaHall;
+import mate.academy.service.CinemaHallService;
+
+@Service
+public class CinemaHallServiceImpl implements CinemaHallService {
+    @Inject
+    CinemaHallDao cinemaHallDao;
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java]: same here

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert MovieSession entity", e);
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save (same goes for other methods with input parameters, where we throw exception)

File: src/main/java/mate/academy/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/model/MovieSession.java:19]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**